flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/UpdatingPlanChecker.scala

```diff
@@ -68,4 +70,18 @@ object UpdatingPlanChecker {
       }
     }
   }
+
+  def getUniqueKeyForUpsertSink(
+      sinkNode: Sink,
+      planner: PlannerBase,
+      sink: UpsertStreamTableSink[_]): Option[Array[String]] = {
+    // extract unique key fields
+    // Now we pick shortest one to sink
+    // TODO UpsertStreamTableSink setKeyFields interface should be Array[Array[String]]
+    val sinkFieldNames = sink.getTableSchema.getFieldNames
+    UpdatingPlanChecker.getUniqueKeyFields(sinkNode.getInput, planner, sinkFieldNames) match {
```

Comment:
`UpdatingPlanChecker.getUniqueKeyFields` is only used by this method. Maybe you can merge it into this method.